components/provisioner/internal/apperrors/apperrors.go

```diff
@@ -9,42 +9,58 @@ const (
 	CodeBadRequest ErrCode = 400
 )
 
+const (
+	IntCodeBadGateway     IntErrCode = 10
+	IntCodeInternal       IntErrCode = 11
+	IntCodeForbidden      IntErrCode = 12
+	IntCodeBadRequest     IntErrCode = 13
+	IntCodeTenantNotFound IntErrCode = 13
+)
+
 type ErrCode int
 
+type IntErrCode int
+
 type AppError interface {
 	Append(string, ...interface{}) AppError
 	Code() ErrCode
+	IntCode() IntErrCode
 	Error() string
 }
 
 type appError struct {
-	code    ErrCode
-	message string
+	code         ErrCode
+	internalCode IntErrCode
+	message      string
 }
 
-func errorf(code ErrCode, format string, a ...interface{}) AppError {
-	return appError{code: code, message: fmt.Sprintf(format, a...)}
+func errorf(code ErrCode, internalCode IntErrCode, format string, a ...interface{}) AppError {
```

Comment:
When we introduce cause field with the following contants:
- Unknown
- TenantNotFound

We will avoid duplicating information as in the following code:
`return errorf(CodeForbidden, IntCodeForbidden, format, a...)`

By default cause field will have Unknown value.